libr/core/project.c

```diff
@@ -677,21 +677,31 @@ R_API bool r_core_project_save(RCore *core, const char *prj_name) {
 		free (prj_bin_dir);
 		free (bin_file);
 	}
-	if (r_config_get_i (core->config, "prj.vc")) {
-		char *git_dir = r_str_newf ("%s" R_SYS_DIR ".git", prj_dir);
-		if (!strcmp ("git", r_config_get (core->config, "prj.vc.type"))
-				&& r_config_get_b (core->config, "prj.vc")) {
-			if (!r_file_is_directory (git_dir)) {
-				r_vc_git_init (prj_dir);
-			}
-			free (git_dir);
-			r_vc_git_add (prj_dir, ".");
-			if (r_cons_is_interactive ()) {
-				r_vc_git_commit (prj_dir, NULL);
-			} else {
-				r_vc_git_commit (prj_dir, "commit");
+	if (r_config_get_b (core->config, "prj.vc")) {
+		if (!rvc_git_repo_exists (core, prj_dir)) {
+			if (!rvc_git_init (core, prj_dir)) {
+				free (prj_dir);
+				free (script_path);
+				return false;
 			}
-
+		}
+		RList *paths = r_list_new ();
+		if (!paths) {
```

Comment:
Be positive and nest instead of having many return statmeents and calling free many times for the same vars in a single function